File: space-game/Data/Classes/OrbitData.vb
Imports Core.Classes

Namespace Classes
    Public Class OrbitData
        Public Property SemiMajorAxis As Distance
        Public Property Eccentricity As Double
        Public Property Inclination As Angle
        Public Property ArgumentOfPeriapsis As Angle
        Public Property LongitudeOfAscendingNode As Angle
        Public Property MeanAnomalyZero As Angle

        ''' <summary>
        ''' Distances and angles must be in the correct units, kilometers and degrees
        ''' </summary>
        Public Sub New(semiMajorAxis As Distance,
                        eccentricity As Double,
                        inclination As Angle,
                        argumentOfPeriapsis As Angle,
                        longitudeOfAscendingNode As Angle,
                        meanAnomalyZero As Angle)
            Me.SemiMajorAxis = semiMajorAxis
            Me.Eccentricity = eccentricity
            Me.Inclination = inclination
            Me.ArgumentOfPeriapsis = argumentOfPeriapsis
            Me.LongitudeOfAscendingNode = longitudeOfAscendingNode
            Me.MeanAnomalyZero = meanAnomalyZero
        End Sub

        Private Sub New ()
            ' For Mapster
        End Sub
    End Class
End Namespace

File: space-game/Data/Classes/OrbitData.cs
using Core.Classes;

namespace Data.Classes
{
    public class OrbitData
    {
        public Distance SemiMajorAxis { get; set; }
        public double Eccentricity { get; set; }
        public Angle Inclination { get; set; }
        public Angle ArgumentOfPeriapsis { get; set; }
        public Angle LongitudeOfAscendingNode { get; set; }
        public Angle MeanAnomalyZero { get; set; }

        /// <summary>
        /// Distances and angles must be in the correct units, kilometers and degrees
        /// </summary>
        public OrbitData(Distance semiMajorAxis, double eccentricity, Angle inclination, Angle argumentOfPeriapsis, Angle longitudeOfAscendingNode, Angle meanAnomalyZero)
        {
            this.SemiMajorAxis = semiMajorAxis;
            this.Eccentricity = eccentricity;
            this.Inclination = inclination;
            this.ArgumentOfPeriapsis = argumentOfPeriapsis;
            this.LongitudeOfAscendingNode = longitudeOfAscendingNode;
            this.MeanAnomalyZero = meanAnomalyZero;
        }

        private OrbitData()
        {
            // For Mapster
        }
    }
}